### PR TITLE
#8415: Prevent the database password from being exported along with the dashboards

### LINKED ISSUE #8415: The database password is exported along with the dashboards
While exporting the dashboards, the database password is exported as well (in plaintext) despite it's not used when importing them later.

### Expected results

The password is not exported at all.

### Actual results

The password is exported.

#### How to reproduce the bug

1. Go to 'Dashboards'
2. Select all dashboards
3. Press Actions -> Export
4. In the downloaded JSON file, look for "password:"

### Environment

(please complete the following information):

- superset version: `0.34.0`
- python version: `3.6.8`
- node.js version: `v8.16.1`
- npm version: `6.11.3`

### Checklist

Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by oskarhinc**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
During export, ehre is a check now if the field is named 'password' and it's not exported if so.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API
- [x] Fixes #8415

### CODE REVIEW

**Comment by oskarhinc** [comment]
Reopening to trigger Travis CI

File: superset/utils/core.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ def __init__(self, *args, **kwargs):
     # pylint: disable=E0202
     def default(self, o):
         try:
-            vals = {k: v for k, v in o.__dict__.items() if k != "_sa_instance_state"}
+            vals = {k: v for k, v in o.__dict__.items() if k != "_sa_instance_state" and k != 'password'}
```
Comment (villebro) [inline superset/utils/core.py]: a cleaner way of doing this would be ...`if k not in ("_sa_instance_state", "password")`

**Review by villebro — COMMENTED** [review]
I tried exporting a dashboard but was unable to reproduce the problem. Which password was this, the user's password?

**Comment by oskarhinc** [comment]
@villebro it's nested in: `datasources -> __SqlaTable__ -> database -> __Database__ -> password`, and it's the same password that is used provided when connecting to the new datasource, which is later replaced with `XXX...` in `sqlalchemy_uri`

File: superset/utils/core.py (hunk 1 above)
Comment (oskarhinc) [reply superset/utils/core.py]: Right, I'll correct it

**Comment by villebro** [comment]
I'm still not able to reproduce this; can you provide any additional context on how this comes about? I only see `"database_id"`, but no nested `database` object. If you create a new dashboard, does it still export the password? Wondering if this is a remnant from some old version..

**Comment by oskarhinc** [comment]
@villebro you're right, it's not occurring for newly created dashboards. I was previously working on version 0.29 and it's possible that I was using this version, while I've noticed exposed password, as I'm not able to reproduce it for the newest version. Although, can we keep this check to ensure it won't be exposed to other cases/configurations?

**Pull request closed without merging**